server/certificate/views.py

```diff
@@ -0,0 +1,90 @@
+from django.http import Http404
+from rest_framework.generics import GenericAPIView
+from rest_framework.response import Response
+from rest_framework import status
+from rest_framework.permissions import IsAuthenticated
+
+from .models import Certificate
+from .serializers import CertificateSerializer
+from .permissions import IsModelApprovedInBenchmark, IsModelOwner
+from user.permissions import IsAdmin
+from drf_spectacular.utils import extend_schema
+from benchmarkdataset.models import BenchmarkDataset
+from dataset.models import Dataset
+from user.permissions import IsOwnUser
+
+
+class CertificateList(GenericAPIView):
+    serializer_class = CertificateSerializer
+    queryset = ""
+
+    def get_permissions(self):
+        """
+        Anyone can post, but only admins can view and edit.
+        Owners can get their own certificates via the CertificateDetail view
+        ModelOwners can get relevant certificates via the CertificatesFromBenchmark view
+        """
+        if self.request.method == "GET":
+            self.permission_classes = [IsAdmin]
+
+    @extend_schema(operation_id="certificates_retrieve_all")
+    def get(self, request, format=None):
+        """
+        List all certificates
+        """
+        certificates = Certificate.objects.all()
+        certificates = self.paginate_queryset(certificates)
+        serializer = CertificateSerializer(certificates, many=True)
+        return self.get_paginated_response(serializer.data)
+
+    def post(self, request, format=None):
+        """
+        Create a new certificate obj
+        """
+        serializer = CertificateSerializer(data=request.data)
+        if serializer.is_valid():
+            serializer.save(owner=request.user)
+            return Response(serializer.data, status=status.HTTP_201_CREATED)
+        return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+
+class CertificateDetail(GenericAPIView):
+    serializer_class = CertificateSerializer
+    queryset = ""
+    permission_classes = [IsAdmin | IsOwnUser]
+
+    def get_object(self, pk):
+        try:
+            return Certificate.objects.get(pk=pk)
+        except Certificate.DoesNotExist:
+            raise Http404
+
+    def get(self, request, pk, format=None):
+        """
+        Retrieve an ca instance.
+        """
+        certificate = self.get_object(pk)
+        serializer = CertificateSerializer(certificate)
+        return Response(serializer.data)
+
+
+class CertificatesFromBenchmark(GenericAPIView):
+    permission_classes = [IsAdmin | (IsModelApprovedInBenchmark & IsModelOwner)]
+
+    def get(self, request, benchmark_id, model_id, ca_id, format=None):
```

Comment:
These two could be replaced by a `IsAssociatedModelOwner` similar to the `IsAssociatedDatasetOwner` permission from `benchmark/permissions.py` I think.